make/env.cmake

```diff
@@ -91,8 +91,12 @@ function(SETUP_ENV)
     set_target_properties(env PROPERTIES USE_CONDA FALSE)
     foreach(binary ${MAYBE_CONDA_BINARIES})
       string(TOUPPER ${binary} binary_upper)
-      set(${binary_upper} ${binary})
+      find_program(${binary_upper} ${binary})
       replace_with_env_if_set(${binary_upper})
```

Comment:
We should probably make this explicit:
if env var -> use env
else use find_program
if not found -> error

Reply:
I've pushed an updated commit that makes this change.